### PR TITLE
[WIP] Add Apigee environment payg node_config support

### PR DESCRIPTION
**Pull request by g-greatdevaks**
Added node_config support for Apigee Evironments which is required for Pay-As-You-Go Apigee Organizations.

modules/apigee-organization has been updated to include a dynamic block for including node_config if billing_type is "PAYG". If billing_type is not "PAYG", node_config block is skipped.

**Note:** Documentation and test cases are to be updated, post which the PR will be marked ready for review.

### CODE REVIEW

**Comment by juliocc** [comment]
@g-greatdevaks is this ready for review?

**Comment by g-greatdevaks** [comment]
@juliocc README.md and some test cases are yet to be updated. Once done. I will mark this as ready for review. Thanks.

File: modules/apigee-organization/variables.tf [hunk 1]
```diff
@@ -43,6 +45,15 @@ variable "apigee_environments" {
     condition     = alltrue([for k, v in var.apigee_environments : contains(["DEPLOYMENT_TYPE_UNSPECIFIED", "PROXY", "ARCHIVE"], v.deployment_type)])
     error_message = "Allowed values for deployment_type \"DEPLOYMENT_TYPE_UNSPECIFIED\", \"PROXY\" or \"ARCHIVE\"."
   }
+
+  validation {
+    condition     = alltrue([for k, v in var.apigee_environments : v.min_node_count >= 2])
```
Comment (juliocc) [inline modules/apigee-organization/variables.tf:50]: can you split in multiple lines. See [here](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/vpc-sc/variables.tf#L90) for example.

File: modules/apigee-organization/main.tf [hunk 2]
```diff
@@ -50,6 +52,15 @@ resource "google_apigee_environment" "apigee_env" {
   deployment_type = each.value.deployment_type
   name            = each.key
   org_id          = google_apigee_organization.apigee_org.id
+
+  dynamic "node_config" {
+    for_each = var.billing_type == "PAYG" ? toset([1]) : toset([])
```
Comment (juliocc) [inline modules/apigee-organization/main.tf:57]: i think the toset() is not needed. We typically use `condition ? []: [1]` ir `condition ? {} : {1=1}`

**Comment by juliocc** [comment]
Closing since this is now obsolete (see #984)

**Pull request closed without merging**